alacritty/src/event.rs

```diff
@@ -919,13 +919,16 @@ impl<'a, N: Notify + 'a, T: EventListener> ActionContext<'a, N, T> {
     fn update_cursor_blinking(&mut self) {
         // Get config cursor style.
         let mut cursor_style = self.config.terminal_config.cursor.style;
-        if self.terminal.mode().contains(TermMode::VI) {
+        let vi_mode = self.terminal.mode().contains(TermMode::VI);
+        if vi_mode {
             cursor_style = self.config.terminal_config.cursor.vi_mode_style.unwrap_or(cursor_style);
-        };
+        }
```

Comment:
Do we need to check for `vi_mode || SHOW_CURSOR`, or can we just check for `cursor_style != HIDDEN`?

Reply:
This is a config cursor style, it can't be hidden. However the cursor style on a terminal could be hidden. And also `Hidden` on a terminal becomes when creating a `RendererCursor`, which is not available here. So we should check like that.